libsolidity/parsing/Parser.cpp

```diff
@@ -2076,16 +2078,39 @@ optional<string> Parser::findLicenseString(std::vector<ASTPointer<ASTNode>> cons
 	for (auto const& [start, end]: sequencesToSearch)
 	{
 		smatch match;
-		if (regex_search(start, end, match, licenseRegex))
+		auto words_begin =
+			std::sregex_iterator(start, end, licenseRegex);
+		auto words_end = std::sregex_iterator();
+
+		for (std::sregex_iterator i = words_begin; i != words_end; ++i)
 		{
-			string license{boost::trim_copy(string(match[1]))};
-			if (!license.empty())
+			std::smatch match = *i;
+			if (!match.empty())
+			{
+				string license{string(match[1])};
+				// Remove trailing comment terminator if present
+				if (boost::algorithm::ends_with(license, "*/"))
+					license = license.substr(0, license.size() - 2);
+
+				license = boost::trim_copy(license);
+
 				matches.emplace_back(std::move(license));
```

Comment:
Those are just strings now. `matches` sounds is if they were still the match objects.
```suggestion
				licenseNames.emplace_back(std::move(license));
```